casper/src/main/scala/coop/rchain/casper/engine/Engine.scala

```diff
@@ -118,6 +117,10 @@ object Engine {
         init,
         disableStateExporter
       )
+      // this is required because before https://github.com/rchain/rchain/pull/3459 genesis as not added to
+      // last finalized storage on network init.
+      _ <- BlockDagStorage[F]
+            .recordDirectlyFinalized(approvedBlock.candidate.block.blockHash, _ => ().pure[F])
```

Comment:
Why this is not already covered by migration? Approved block will be migrated through parents traversal. Am I missing something?

Reply:
When new network is started, or node is bootstrapped from LFB - approved block has to be marked as directly finalized.

Reply:
Received LFB is inserted as approved in Initializing engine.
https://github.com/rchain/rchain/blob/1370ee414b4336fc273b3d6d73014666dc331330/casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala#L100-L101

And this is called by genesis master.
https://github.com/rchain/rchain/blob/f2c030b0a906e26343211daad214411bb0add306/casper/src/main/scala/coop/rchain/casper/engine/Engine.scala#L63-L71

Is there some other case?

Reply:
If network is started and no finalized block has been found, genesis is not recorded as finalized. On switching to this migration build there will be no finalized block.

Reply:
Sorry, but I still don't understand why not fixed this as part of migration. We have approved block in BlockStore and we can add it to BlockMetaData store.

Reply:
Makes sense, https://github.com/rchain/rchain/pull/3459/commits/cf6106b2b00ee6687706e192bc62876ca429bc61